composables/apis/useTorDetails.ts

```diff
@@ -0,0 +1,92 @@
+import { ref, Ref } from "vue";
+
+interface FetchError {
+  message: string;
+  status?: number;
+}
+
+export const useTorDetails = () => {
+  const data: Ref<any | null> = ref(null);
+  const error: Ref<FetchError | null> = ref(null);
+  const isLoading: Ref<boolean> = ref(false);
+
+  const API_KEY = 'cc09f9516dea4a9e804f6885921f231f';
+
+
+  const getLatLonFromIPs = async (ips: string[]): Promise<any[]> => {
+      try {
+          const response = await $fetch('https://api.ipgeolocation.io/ipgeo-bulk?apiKey=cc09f9516dea4a9e804f6885921f231f', {
+              method: 'POST',
+              headers: {
+                  'Content-Type': 'application/json'
+              },
+              body: JSON.stringify({ ips })
+          });
+
+          if (response && Array.isArray(response)) {
+              return response;
+          } else {
+              throw new Error(`Unexpected response structure from IP geolocation API.`);
+          }
+      } catch (error) {
+          console.error('Error fetching IP coordinates:', error);
+          return [];
+      }
+  }
+
+  const fetchDetails = async () => {
+    isLoading.value = true;
+    error.value = null;
+
+    try {
+      const responseData: any = await $fetch(
+        "https://onionoo.torproject.org/details"
```

Comment:
1. Similar to the comment above, this introduces a new, centralized, third-party dependency directly to the dashboard.  Yes, technically the `validator` currently looks at Onionoo, but this may not always be the case.  This also may not scale if hundreds of users are requesting simultaneously.  This also exposes user IP addresses to Onionoo.
2. Again, similar to the comment above, this could be part of the `validator` and published as a new metrics entity type.